tests/unit/socket_test.cc

```diff
@@ -258,3 +261,57 @@ SEASTAR_TEST_CASE(socket_connect_abort_test) {
         when_all(std::move(cf), std::move(check), std::move(abort)).get();
     });
 }
+
+SEASTAR_THREAD_TEST_CASE(socket_bufsize) {
+
+    // Test that setting the send and recv buffer sizes on the listening
+    // socket is propagated to the socket returned by accept().
+
+    auto buf_size = [](std::optional<int> buf_size) {
+        listen_options lo{
+            .reuse_address = true,
+            .lba = server_socket::load_balancing_algorithm::fixed,
+            .so_sndbuf = buf_size,
+            .so_rcvbuf = buf_size
```

Comment:
Nitpick: Why did you decide to check this function with just one parameter (buf_size) used for both sndbuf and rcvbuf? If you had a copy-pasto in the implementation where SNDBUF was taken from the rcvbuf parameter, your test wouldn't catch it.
I think it will be better to pass two options, sndbuf and rcvbuf, to this test function, and pass different sizes to it - not just two identical small and two identical large.

Reply:
Good idea, I've updated this to use separate sizes in https://github.com/scylladb/seastar/commit/5b33a115045eeb037ff41db9c22c213e45808508.